group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1001,8 +1002,26 @@ public void onTransactionCompleted(
     public void onPartitionsDeleted(
         List<TopicPartition> topicPartitions,
         BufferSupplier bufferSupplier
-    ) {
+    ) throws ExecutionException, InterruptedException {
```

Comment:
To clarify -- is onPartitionsDeleted only called when the topic behind the partition is deleted (not reassigned etc)? Or are there other cases?

Reply:
That's right. It is only called for deleted topics. The logic for this is in the `BrokerMetadataPublisher` in case you want to take a look.